src/steamship/agents/schema/agent.py

```diff
@@ -57,4 +57,20 @@ def messages_to_prompt_history(messages: List[Block]) -> str:
                 as_strings.append(f"System: {block.text}")
             elif role == RoleTag.AGENT:
                 as_strings.append(f"Agent: {block.text}")
+            elif role == RoleTag.FUNCTION:
+                as_strings.append(f"Function: {block.text}")
         return "\n".join(as_strings)
+
+
+# TODO(dougreid): does this need to be a separate thing? or should we just use LLM agent, but with a ConversationalLLM.
+class ConversationalLLMAgent(Agent):
+    """ConversationalLLMAgents choose next actions for an AgentService based on chat-based interactions with an LLM."""
+
+    llm: ConversationalLLM
+
+    output_parser: OutputParser
+    """Utility responsible for converting LLM output into Actions"""
+
+    @abstractmethod
+    def next_action(self, context: AgentContext) -> Action:
```

Comment:
Does this redefinition of the abstract method do something?

Reply:
nothing other than saying "yup, this still needs to be implemented". I don't know what the pythonically-appropriate way of marking that is.

Reply:
If you mark ConversationalLLMAgent as still inheriting from ABC, can you just omit it?

Reply:
done.